File: solutions/415.add-strings/add-strings.java
class Solution {
    public String addStrings(String num1, String num2) {
        int carry = 0;
        String res = "";
        int len1 = num1.length(), len2 = num2.length();
        for(int i = 0; ;i++){
            if(i >= len1 && i >= len2){
                if(carry != 0)
                    res = carry + res;
                return res;
            }
            int item1 = 0, item2 = 0;
            if(i < len1){
                item1 = num1.charAt(len1 - 1 - i) - '0';
            }
            if(i < len2){
                item2 = num2.charAt(len2 - 1 - i) - '0';
            }
            res = (item1+item2+carry) % 10 + res;
            carry = (item1+item2+carry) / 10;
        }
    }
}

File: solutions/415.add-strings/add-strings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var addStrings = function(num1, num2) {
    let res = '';
    let carry = 0;
    let i = num1.length - 1, j = num2.length -1;
    while(i >= 0 || j >= 0  || carry !== 0){
        let digit1 = i >= 0 ? +num1[i--] : 0;
        let digit2 = j >= 0 ? +num2[j--] : 0;
        const sum = digit1 + digit2 + carry; 
        res =  (sum % 10) +res ;
        carry = parseInt(sum / 10, 10);
    }
    return res;
};